In [3]:
import pandas as pd
dataset=pd.read_csv("general_data.csv")
dataset.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [4]:
dataset.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [9]:
dataset.duplicated

<bound method DataFrame.duplicated of       Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1          

In [10]:
dataset.drop_duplicates()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [15]:
a_yes=dataset[dataset['Attrition']=='Yes']
a_no=dataset[dataset['Attrition']=='No']

In [20]:
from scipy.stats import ttest_ind
stats,p=ttest_ind(a_yes.Age, a_no.Age)
print(stats,p)

-10.70661860061757 1.9968016158894556e-26


__Hypothesis 1 Age v/s Attrition__

__TTest__

H0(Null Hypothesis):- There is no significance difference between the Age with respect to Attrition

Ha(Alternative Hypothesis):- There is a significance difference between the Age with respect to Attrition

p value :- 1.9968016158894556e-26 , which is less than 0.05

__Reject H0__

__Conclusion :- There is a significance difference between the Age with respect to Attrition__

In [38]:
from scipy.stats import mannwhitneyu
stats,p=mannwhitneyu(a_yes.MonthlyIncome, a_no.MonthlyIncome)
print(stats,p)

1264900.5 0.053577283839938566


__Hypothesis 2 Monthly Income v/s Income__

__MannwhitneyTest__

H0(Null Hypothesis):- There is no significance difference between the Age with respect to MonthlyIncome

Ha(Alternative Hypothesis):- There is a significance difference between the Age with respect to MonthlyIncome

p value :- 0.053577283839938566 , which is greater than 0.05

__Accept H0__

__Conclusion :- There is no significance difference between the Age with respect to MonthlyIncome__

In [39]:
from scipy.stats import mannwhitneyu
stats,p=mannwhitneyu(a_yes.YearsAtCompany, a_no.YearsAtCompany)
print(stats,p)

923238.0 6.047598261692858e-37


__Hypothesis 3 YearsAtCompany v/s Attrition__

__MannwhitneyTest__

H0(Null Hypothesis):- There is no significance difference between the YearsAtCompany with respect to Attrition

Ha(Alternative Hypothesis):- There is a significance difference between the YearsAtCompany with respect to Attrition

p value :- 6.047598261692858e-37 , which is less than 0.05

__Reject H0__

__Conclusion :- There is a significance difference between the YearsAtCompany with respect to Attrition__

In [40]:
stats,p=mannwhitneyu(a_yes.TotalWorkingYears, a_no.TotalWorkingYears)
print(stats,p)

909954.0 2.8897554531400447e-39


__Hypothesis 4 TotalWorkingYears v/s Attrition__

__MannwhitneyTest__

H0(Null Hypothesis):- There is no significance difference between the TotalWorkingYears with respect to Attrition

Ha(Alternative Hypothesis):- There is a significance difference between the TotalWorkingYears with respect to Attrition

p value :- 2.8897554531400447e-39 , which is less than 0.05

__Reject H0__

__Conclusion :- There is a significance difference between the TotalWorkingYears with respect to Attrition__

In [41]:
stats,p=ttest_ind(a_yes.DistanceFromHome, a_no.DistanceFromHome)
print(stats,p)

-0.6460416038042738 0.518286042805572


__Hypothesis 5 DistanceFromHome v/s Income__

__TTest__

H0(Null Hypothesis):- There is no significance difference between the Age with respect to DistanceFromHome

Ha(Alternative Hypothesis):- There is a significance difference between the Age with respect to DistanceFromHome

p value :- 0.518286042805572 , which is greater than 0.05

__Accept H0__

__Conclusion :- There is no significance difference between the Age with respect to DistanceFromHome__

In [42]:
from scipy.stats import chi2_contingency
chitable=pd.crosstab(dataset.YearsAtCompany , dataset.MonthlyIncome)
stats,p,dof,expected=chi2_contingency(chitable)
print(stats,p)

145781.2864837561 0.0


__Hypothesis 6 YearsAtCompany v/s MonthlyIncome__

__ChiSquareTest__

H0(Null Hypothesis):- There is no significance difference between the YearsAtCompany with respect to MonthlyIncome

Ha(Alternative Hypothesis):- There is a significance difference between the YearsAtCompany with respect to MonthlyIncome

p value :- 0.0 , which is less than 0.05

__Reject H0__

__Conclusion :- There is a significance difference between the YearsAtCompany with respect to MonthlyIncome__